packages/@lwc/style-compiler/src/serialize.ts

```diff
@@ -165,21 +186,44 @@ function serializeCss(result: Result, collectVarFunctions: boolean): string {
             currentRuleTokens.push({ type: TokenType.text, value: part });
 
             // If we are in fakeShadow we dont want to have :host selectors
-            if ((node as any)._isHostNative) {
+            if ((node as any)._isNativeHost) {
                 // create an expression for all the tokens (concatenation of strings)
                 // Save it so in the next rule we can apply a ternary
-                tmpHostExpression = generateExpressionFromTokens(currentRuleTokens);
-            } else if ((node as any)._isFakeNative) {
-                const exprToken = generateExpressionFromTokens(currentRuleTokens);
+                nativeHostTokens = [...currentRuleTokens];
+            } else if ((node as any)._isSyntheticHost) {
+                if (!nativeHostTokens) {
+                    throw new Error('Unexpected host rules ordering');
+                }
```

Comment:
Like the same error below, this should never be hit.

Reply:
Could you add `/* istanbul ignore if */` on these?